ivy.el

```diff
@@ -1965,14 +1965,12 @@ This is useful for recursive `ivy-read'."
          (let ((w (length (number-to-string
                            (length ivy--all-candidates))))
                (s (copy-sequence ivy-count-format)))
-           (string-match "%d" s)
-           (match-end 0)
```

Comment:
The call to `match-end` is indeed a no-op, but I'm not sure you can get rid of the call to `string-match`.

Reply:
`string-match` here is followed by another `string-match`, which would overwrite match data.

Reply:
@sergv 
> `string-match` here is followed by another `string-match`, which would overwrite match data.

Yes, but the second `string-match` uses the `match-end` of the first `string-match` as its starting index.

Reply:
I have found a way to replace two `string-match`es with a single one. Please take a look at `Simplify ivy-add-prompt-count` commit.

Reply:
@sergv 

> I have found a way to replace two `string-match`es with a single one.

Nice, but why stop there? :)

```diff
diff --git a/ivy.el b/ivy.el
index 193af0a..d154bbf 100644
--- a/ivy.el
+++ b/ivy.el
@@ -1987,17 +1987,11 @@ ivy-add-prompt-count
         ((null ivy-count-format)
          (error
           "`ivy-count-format' can't be nil.  Set it to \"\" instead"))
-        ((string-match "%d.*%d" ivy-count-format)
-         (let ((w (length (number-to-string
-                           (length ivy--all-candidates))))
-               (s (copy-sequence ivy-count-format)))
+        ((string-match "%d.*\\(%d\\)" ivy-count-format)
+         (let* ((w (1+ (floor (log (max 1 (length ivy--all-candidates)) 10))))
+                (s (replace-match (format "%%-%dd" w) t t ivy-count-format 1)))
            (string-match "%d" s)
-           (match-end 0)
-           (string-match "%d" s (match-end 0))
-           (setq s (replace-match (format "%%-%dd" w) nil nil s))
-           (string-match "%d" s)
-           (concat (replace-match (format "%%%dd" w) nil nil s)
-                   prompt)))
+           (concat (replace-match (format "%%%dd" w) t t s) prompt)))
         ((string-match "%.*d" ivy-count-format)
          (concat ivy-count-format prompt))
         (ivy--directory
```

This way, there is less searching, way less string allocation, and you also get to fix the `replace-match` arguments `FIXEDCASE` and `LITERAL` which shouldn't have been `nil`.